flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotMatchingStrategy.java

```diff
@@ -41,4 +42,18 @@ <T extends TaskManagerSlotInformation> Optional<T> findMatchingSlot(
             ResourceProfile requestedProfile,
             Collection<T> freeSlots,
             Function<InstanceID, Integer> numberRegisteredSlotsLookup);
+
+    /**
+     * Finds a matching resource instance for the request {@link ResourceProfile} given the
+     * collection of available instances.
+     *
+     * @param resourceMatchingPredicate Predicate of whether instance resource is matching
+     * @param availableInstances collection of available instances
+     * @param numberUtilizationLookup lookup for the weight of instance, lower is better
+     * @return Returns a matching instance or {@link Optional#empty()} if there is none
+     */
+    Optional<InstanceID> findMatchingResource(
+            Predicate<InstanceID> resourceMatchingPredicate,
+            Collection<InstanceID> availableInstances,
+            Function<InstanceID, Double> numberUtilizationLookup);
```

Comment:
Why `numberUtilizationLookup` is required for the interface. IIUC, only `LeastUtilizationSlotMatchingStrategy` need this lookup function. It's strange to put it in the interface protocol.